src/librustc/session/config.rs

```diff
@@ -515,6 +515,8 @@ options! {CodegenOptions, CodegenSetter, basic_codegen_options,
         "explicitly enable the cfg(debug_assertions) directive"),
     inline_threshold: Option<usize> = (None, parse_opt_uint,
         "set the inlining threshold for"),
+    disable_gold: bool = (false, parse_bool,
```

Comment:
> We already have -C linker which possibly swaps out gcc for something else and now we pass a special flag into linker assuming it is gcc-compatible (in absence of this option) without checking if specified linker is indeed compatible. That might make -C linker harder to use

As far as I can tell that is an existing problem. Even if you override the linker, rustc assumes it takes either `gcc` or `link.exe` style arguments, depending on platform. IOW I tihnk `-C linker` is only good for specifying a different path to the expected linker, not for using a completely different 'style' of linker.